### DISCUSSION TITLE
Tactor Specs

### DISCUSSION THREAD

**Discussion by TactileDesign** [kriswilk/buzzah#24] [General]
Does anyone have a tactor file they will share for use in a 3D printer?  Also, where is the diameter of the contact area documented?  I think I remember reading Dr. Tass called for 5mm diameter, but I cant seem to find the reference anywhere....

**Comment by Neurosmith24** [comment]
I've got several versions of the tactor housing that I've created that I could share. But, I use a different LRA than the original one here. Which Vybronics LRA do you use?

**Reply by TactileDesign** [reply]
Hello Neurosmith24,  I think I am using the same LRA as you: VLV101040A (the square).   Could you also recommend a 3D printer which could be used?  And what kind of CAD program are you using?  I am just getting into the world of 3D printing, so I have a lot to learn.  BTW, you will probably be interested to see the bench test results I ran for 3 LRAs.  I found that the VLV101040A performed much better than its specs thanks to our driver chip that has software enhanced starting and braking.  (Note:  The bench tests make up a small part of a pdf I am creating describing my build that is housed on a bluetooth neck speaker)

**Comment by Neurosmith24** [comment]
Ok, I've got a couple of different designs for that LRA. I thought I had a file I could easily share but looks like I need to find it or recreate it. 
For the CAD program I've been using a free online CAD tool from AutoDesk called TinkerCAD. I'd suggest starting there. 

For 3D printer, I've been getting them from our local library's maker space, but their hours are sporadic at best so I just ordered a 3D printer a few days ago that arrives tomorrow. This is the one I went with:

https://www.amazon.com/gp/product/B0CN6RX6BJ/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&psc=1

Great reviews from what I've read and pretty affordable as 3D printers go. I'll see about getting you some .stl files of my housings this weekend and sharing them here.

**Reply by TactileDesign** [reply]
Thanks for the info!  I will view you .stl's with great interest on TinkerCAD.   Were you able to dampen the vibration of the housing sufficiently such that vibrations are localized on the contact area?  Also, where did you find a dimension for your fingertip contact area?

**Reply by Neurosmith24** [reply]
<img width="947" alt="Screenshot 2024-09-21 at 11 03 26 AM" src="https://github.com/user-attachments/assets/0b11cc0e-04ec-4d17-8422-10c03ecb1e86">
I was able to dampen the non-z-axis vibration some with some foam padding. I put a bit of 3mm foam padding underneath the LRA and then some surrounding it. This image is from TinkerCad. The purple bits represent the foam. The yellow bits represent the LRA, the  beige thing on top of the LRA is the focusing contact point - 5 mm on top, 8 mm bottom part.

This is image just shows that actual 3D printed housing:
<img width="961" alt="Screenshot 2024-09-21 at 11 04 24 AM" src="https://github.com/user-attachments/assets/26790f73-5798-4bae-a8b7-7c28595ce4c7">
The lighter blue is where the finger sits. The darker blue is where the LRA is installed. You'll need 4 small screws to attach the finger plate to the housing base. I used these screws: 

https://www.amazon.com/dp/B000NHYQ7S?ref=ppx_yo2ov_dt_b_fed_asin_title&th=1

The 5 mm dimensions are referenced in a Tass paper - I believe the 2017 paper but I'm not positive on that. 

Doesn't look like I can upload a .stl file. Let me know when you register for TinkerCAD what your username is and I can try to share via that.

**Reply by TactileDesign** [reply]
OK!  thanks!  I created my Tinkercad account under the same username as here:  TactileDesign

**Reply by Neurosmith24** [reply]
Looks like I can just share a link. Try this:
https://www.tinkercad.com/things/c5nAxhal8g1-tactor-housing-redesign-for-sharing/edit?returnTo=%2Fdashboard%2Fdesigns%2F3d&sharecode=j4o4aHCE-5jZ1Ol_BFWp66Tavzqy1_hvxmfYdnH1oLc

**Comment by TactileDesign** [comment]
Thanks for the link!  A few questions:
1.  I don’t see the little button that sits on top of the buzzer.  Is that in a different file?
2.  Is the button glued to the top of the buzzer with crazy glue?
3.  What kind of foam did you use?
4.  How did you cut the foam so precisely?
5.  Is the foam glued in or does it just sit in the enclosure?
6.  It looks to me like the only thing that prevents the buzzer from coming into contact with the underside of finger rest plate is the fact that the buzzer is becomes depressed downward by the forces of the finger pushing down on the button.  Is that correct?
7.  I saw on Github where you initially had trouble getting vibrations to be focused at the fingertip.  How was this problem finally resolved?
And a couple of rookie questions:  
8.  The housing appears to extend below the printer’s assembly plane.  Does the printer automatically adjust the itself to avoid this physical impossibility?
9.  How does the printer know not to print the images of the buzzer and foam that appear in the picture?  Did you insert those as "nonprint objects"?

**Reply by Neurosmith24** [reply]
I'll respond to a couple of these quickly but a full response may come later. 

I just added buttons to the file. I have different sizes there to try out. I started with the shortest but adjusted to taller ones as needed. 

The button is secured with a piece of double-sided tape stuck to the LRA top. I wanted to be able to swap out contact points so I didn't use glue.

There should be enough gap between the LRA and the bottom of the finger plate that it doesn't contact it. 

I just moved the housing up above the plane. I had it lower while I was working with it just to help me make calculations.

When you're ready to print. Delete the objects representing the foam, the LRA, etc. Maybe copy the file to have a separate version that's just the housing bits. What I do is make a separate file with everything I want printed. Like this:

<img width="1306" alt="Screenshot 2024-09-25 at 8 56 28 AM" src="https://github.com/user-attachments/assets/78b979e4-b399-4dd2-ba3f-d01443eed73c">

I fact, I'll make a copy of that file and share it with you and you can just use that.

https://www.tinkercad.com/things/26b9Er2vdZk-copy-of-wtp-tactor-housing-redesign-v1-for-print/edit?sharecode=SM815ma0zf1BpuR9xy4ELv_2KZUO2adWWL4BUR8EXng

**Comment by Neurosmith24** [comment]
Ok, continuing on to other questions. Here's the foam I used:

https://www.amazon.com/dp/B0BJKLY8RD?ref=ppx_yo2ov_dt_b_fed_asin_title

To cut it I used an exacto knife and scissors. It's fussy and difficult to get it perfect, especially the second layer. The foam comes with an adhesive. The first few I peeled off the backing, but later ones I didn't bother. 

I think the foam did help somewhat in focusing the vibrations on the fingertip but not a lot. The reality is that there's still a lot of general buzzing through the housing to the whole finger. 

I did find that some LRAs performed stronger and seemed more focus than others. I initially thought that I had a few good LRAs and the others were just kinda poor. So I ordered some more. And initially they were strong too. But I noticed a few weeks ago that some LRAs felt noticably weaker and less focused, and then even had some that were initially really strong and focused, stop working completely ( I swapped them out for older ones I had and they worked so it was an LRA issue not a driver or circuit issue)

As a result in the frustration of the poor reliability we've had with the LRAs, several inexplicable glitches over the months, along with the lack of focus of vibration on the contact point, I've actually started making a new set of gloves that will use audio exciters based on the system developed and documented [here](https://www.curepd2.com/) and [here](https://healthunlocked.com/cure-parkinsons/posts/150536169/winnie-the-poos-glove-building-adventure).

I think core electronics of the Buzzah system is fantastically simple and elegant, but in the end, I just don't feel like LRAs are the way to best get the desired vibration.

**Comment by TactileDesign** [comment]
Thanks for all the detailed info you shared. I hope you can get your LRAs to work better for you.  It is only through sharing experiences like this that we can really move forward.  Your problems forced me to really think deeply about the LRA tactor challenge which is fundamentally different for acoustic exciters.    If you end up building an acoustic exciter system I would be interested to hear how that works for you...  

Regarding your LRA performance issues, my experience has been very different.  All 8 of my motors vibrate vigorously – giving my fingertips good zingy feeling for every buzz.  The buzzes I feel are actually a bit too vigorous and I have been thinking I need to go in and adjust the amplitude downward in Kris’s settings.  

I have been using my gloves for at least 4 hours a day for over 3 months and I have had only 1 go bad.  For the one that did go bad, it was as you described – it didn’t stop working altogether, but rather, it was still buzzing, but only very weakly.  It went bad in the first week and I worried I was going to be forever replacing bad motors.  But I have had no trouble since with any others.  I attribute the bad motor I had to two possibilities:  1) It had a manufacturer defect or 2) In my early testing of performance, I really cranked up the voltage to see what they could do and I had my amplitude set higher than 2.5.  I also had my amplitude max at 127.  I ran all 8 buzzers like that until the one failed – I then replaced the bad one and backed down voltage to 2.5 and amplitude to 100 and have not had a problem.  

Regarding your comment about “inexplicable glitches” -  I think I have also observed this a few times – maybe 3 times in 3 months.  What happens is that, instead of playing the tass buzz sequence, all motors will turn on together and not stop.  When that happens, I just restart my Buzzah (by turning off and back on) and everything is fine again.  I think this happens because Kris has our i2Q buss frequency set to maximum - pushing the hardware to operate at its upper limit.  I think it is actually way higher than we need – given the response times our buzzes need.  I thought to go in and see if I could set the buss frequency a bit lower, but since the glitches happen so infrequently, it seems easier to just reset the Buzzah once in a month. 

I am thinking that the differences we observe in performance might be due to how we are both using our buzzers.  Here are a few follow-up questions 

1.  What settings do you have indicated in the code?  Here are Kris’s stock settings:
ACTUATOR_TYPE = "LRA"
ACTUATOR_VOLTAGE = 1.8 * math.sqrt(2)
ACTUATOR_FREQUENCY = 235
AMPLITUDE_MIN = 100
AMPLITUDE_MAX = 100
TIME_ON = 0.100
TIME_OFF = 0.067
TIME_RELAX = 4 * (TIME_ON + TIME_OFF)
JITTER = 23.5
MIRROR = False
TIME_SESSION = 120

2.  How do you physically test a buzzer?  Do you test it in your tactor housing, or is it freely vibrating in your hand?

3.  What process did you go through when selecting your foam?  Did you experiment with different degrees of stiffness?  

Thoughts on the physics of z-axis buzzer vibration
When you have your motor in a housing, you are basically pinching it between two springs.  Ideally these two springs would each have very low stiffness to allow for the greatest vibrational amplitude of the motor box – it is this displacement of the motor box that gives us our vCR zingy buzz feeling as the vibrations make our skin surface go up and down.

One of the two springs holding the motor is your skin – which would have very low firmness as long as you don’t have the Velcro tighteners too tight.  And, the smaller finger contact area we have, the less resistance our skin will present as noted in one of Dr tass's papers.  Since I don’t have any topper on my motors to minimize area to 5mm like you do, my area of contact is greater – so my “skin spring” is much stiffer than yours. So your skin spring should be better than mine.

The other spring is the foam below the motor, which I think needs to be as cushy as possible.  If the foam is too stiff, then the motor is not free to vibrate up and down because, in order to do so, it must move the entire tactor housing.  

Ultimately, the motors housed in your tactors should vibrate much stronger than mine, since your skin contact area is smaller making your “skin spring” less stiff – and your foam on the other side could be like a marshmello, offering less stiffness even than the Velcro tighteners that make up my bottom spring.   

Note:  Your topper, that has the 5mm contact area, adds a small mass between your finger and motor.  This dampens the motion of the motor box a bit since – since, in order to move, the motor box has to also move the topper.  Some additional vibrational amplitude might be gained by making your topper hollow- or otherwise less massive.   (When you test your buzzers, is your topper already affixed to the top of your motor?)

Another thing I will add is that, in my testing of 3 LRAs from Digikey, I found the one we are both using to clearly have the greatest vibrating amplitude – but there is a caveat.  I observed it has two distinct modes of vibration.  The first mode I will call “healthy mode” is observed when the box is allowed to vibrate loosely held between two fingers that act like two weak springs.  (So the vibration of the motor we feel in “healthy” mode comes from the back and forth motion of the entire motor box moving in space)

The other mode I will call “choked mode” happens if the motor’s external vibration is stifled by holding it against a larger mass – so one of the two ideal low stiffness springs is essentially replaced by a high stiffness spring.  I can easily observe the choked mode by setting a motor on a table top and pushing down on it with my finger.  When a motor is forced into “choked” mode, I can hardly feel the vibration – I suspect that is what you might be getting. (Note: The vibration of the motor I feel in “choked” mode does not comes from the back and forth motion of the entire motor box moving in space – since the box is no longer free to vibrate.  Instead, the vibration I feel in choked mode comes from the tiny vibrational motion of the actual motor box top made by the sound of the internal vibration – it is like listening to a conversation going on in another room – any sound waves that we can hear outside the room have to come from the actual motion of the wall)

So the upshot is, I think your foam may be too stiff and it is forcing your motors to vibrate in choked mode. How firm is your foam?  Does it depress more like a marshmallow, or is it much stiffer?   My guess from reading Amazon reviews is that your foam is pretty stiff – which may be the reason you are getting the vibrations transferred to your housing.  

4.   I think I read in one of your posts that you were buzzing around 170 Hz (the motor’s resonant frequency) to get more amplitude.  Is that right?  If so, this could be the key difference in our longevity experiences. I buzz at Tass’s 250 HZ, which gives a much lower amplitude – but still more amplitude that I need for vCR.  If you are buzzing around the resonant frequency, your motors are getting a much stronger work out than mine since internal the vibrational amplitude is so much higher at that frequency - this could be happening in "chocked mode" even when you dont feel a large external vibration - so the internal motion can still be high inside the motor box even though the motor box is fixed.  Likewise, I would expect the lifetime to be much shorter for your motors since you are working them so much harder.

Thanks again!

**Comment by kriswilk** [comment]
Without personal experience building the system out into a complete set of gloves, I've just been keenly observing. Thanks for both of your contributions in this thread, @Neurosmith24 and @TactileDesign!

@TactileDesign, I agree that you definitely don't want the actuators rigidly affixed to anything with significant mass. If you do, the movement of the actuator casing (which is what imparts vibrational energy to your fingertip) will be significantly attenuated.

As for the longevity of the LRAs, I can't comment authoritatively. But I suspect that, even within the operating range of an LRA, using it regularly for such a long time is probably beyond the anticipated usage pattern. Being more conservative about the driving parameters (if possible) probably does help to extend their lifetime.
